### PR TITLE
Enable Publishing to Ballerina Central on Release

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose
- Improve the build script
- Enable publishing to Ballerina Central when `gradle publish` is called
- Bump the version to `<next_minor_version>-alpha2-SNAPSHOT`
- Remove unnecessary gradle tasks

### CODE REVIEW

File: os-ballerina/build.gradle [hunk 1]
```diff
@@ -255,7 +250,7 @@ publishing {
     repositories {
         maven {
             name = "GitHubPackages"
-            url = uri("https://maven.pkg.github.com/ballerina-platform/module-ballerina-os")
+            url = uri("https://maven.pkg.github.com/ballerina-platform/module-${packageOrg}-${packageName}")
```
Comment (BuddhiWathsala) [inline os-ballerina/build.gradle:253]: I don't see any advantage of parameterizing this GitHub URL. Even for some modules, the URL and the `packageOrg` are different. Therefore, we have to be careful when we change them like this.

Comment (ThisaruGuruge) [reply os-ballerina/build.gradle:253]: How can the URL and the packageOrg is different? That's weird. The reason to add this here is to remove the hard-coded strings as much as possible.

File: os-ballerina/build.gradle [hunk 2]
```diff
@@ -137,60 +138,51 @@ task initializeVariables {
...
-    dependsOn(initializeVariables)
-    finalizedBy(revertTomlFile)
-
     doLast {
         exec {
             workingDir project.projectDir
             environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
             if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat test --code-coverage --includes=* ${groupParams} ${disableGroups} ${debugParams}"
+                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat test --code-coverage --includes=* ${groupParams} ${disableGroups} ${debugParams}  && exit %%ERRORLEVEL%%"
```
Comment (MadhukaHarith92) [inline os-ballerina/build.gradle:166]:
```suggestion
                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat test --code-coverage --includes=* ${groupParams} ${disableGroups} ${debugParams}"
```

Comment (MadhukaHarith92) [reply os-ballerina/build.gradle:166]: Tests are failing in Windows, need to fix https://github.com/ballerina-platform/ballerina-standard-library/issues/160

**Review by MadhukaHarith92 — APPROVED** [review]

File: os-ballerina/build.gradle [hunk 3]
```diff
@@ -160,37 +171,18 @@ task ballerinaTest {
...
 
     doLast {
         // Build and populate caches
         exec {
             workingDir project.projectDir
             environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
             if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat build " +
-                        "$testParams ${debugParams}"
+                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat build ${testParams} ${debugParams}  && exit %%ERRORLEVEL%%"
```
Comment (MadhukaHarith92) [inline os-ballerina/build.gradle:183]:
```suggestion
                commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat build ${testParams} ${debugParams} "
```

**Pull request merged**